### PR TITLE
Move IovArray handling code in an extra class to better seperate it and

### PR DESCRIPTION
**Pull request by normanmaurer**
easier to test.

Motivation:

We should move the IovArray related code to an extra class so its easier
to test

Modifications:

- Move into extra class
- Add dedicated test

Result:

Cleanup

### CODE REVIEW

**Review by chrisvest — APPROVED** [review]

**Review by njhill — APPROVED** [review]
LGTM, though I'm curious why there is an array of arrays rather than just one x8 the length which is sliced up

**Comment by normanmaurer** [comment]
Because it was easier to just reuse this for now ;)

File: transport-native-io_uring/src/main/java/io/netty/channel/uring/IovArrays.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+            }
+        }
+        return iovArray.isFull() ? null : iovArray;
+    }
+
+    /**
+     * Clear all {@link IovArray}s that were used since the last time this method was called.
+     */
+    void clear() {
+        for (int i = 0; i <= iovArrayIdx; i++) {
```
Comment (1Jo1) [inline transport-native-io_uring/src/main/java/io/netty/channel/uring/IovArrays.java:52]: should be `< `instead of` <=` right?

Comment (1Jo1) [inline transport-native-io_uring/src/main/java/io/netty/channel/uring/IovArrays.java:52]: should be < instead of <= right?

**Review by 1Jo1 — CHANGES_REQUESTED** [review]
looks good :) just one comment

**Comment by njhill** [comment]
I think the existing class would already support that if capacity could be specified in the constructor? e.g. something like
```java
int count = iovArray.count();
// add buffers to iovArray
long iovAddress = iovArray.memoryAddress(count);
int iovLen = iovArray.count() - count;
```

**Comment by 1Jo1** [comment]
> I think the existing class would already support that if capacity could be specified in the constructor? e.g. something like
> 
> ```java
> int count = iovArray.count();
> // add buffers to iovArray
> long iovAddress = iovArray.memoryAddress(count);
> int iovLen = iovArray.count() - count;
> ```

but the array capacity is only 1024 bytes

File: transport-native-io_uring/src/main/java/io/netty/channel/uring/IovArrays.java (hunk 1 above)
Comment (normanmaurer) [reply transport-native-io_uring/src/main/java/io/netty/channel/uring/IovArrays.java:52]: Nope as this is the idx

**Comment by normanmaurer** [comment]
@njhill its not quite as easy... I will look into it tho. But lets merge this one first.

**Review by 1Jo1 — APPROVED** [review]

**Pull request merged**